pkgs/development/ocaml-modules/janestreet/janePackage.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, ocaml, jbuilder, findlib }:
+
+{ name, version ? "0.9.0", buildInputs ? [], hash, ...}@args:
+
+if !stdenv.lib.versionAtLeast ocaml.version "4.03"
+then throw "${name}-${version} is not available for OCaml ${ocaml.version}" else
+
+stdenv.mkDerivation (args // {
```

Comment:
In general yes, but `janePackage` is not supposed to be a generic builder, it is fitted to a very homogeneous set of packages (and in practice it has worked well). Doing what you suggest would prevent changing the value of `buildInputs` as is done currently. If in the future, we want to make some more attributes configurable it can be done by using an argument with a default value (as for `version` or an `if`-test where this attribute is defined).